news/_posts/2019-09-11-prestashop-in-2019-and-beyond-part-2-pain-points.md

```diff
@@ -0,0 +1,471 @@
+{: .text-center style="margin: 5rem auto"}
+
+In the [previous part][part-1], we described what the current architecture looks like. This article aims to analyze what are the main "pain points", that is, fundamental problems that are dragging the project down.
+
+But before getting to that, let's address the elephant in the room. After reading the previous part in this series, probably the number one thing you thought of was "why _the hell_ is this so complicated?". 
+
+There are many reasons for that, but I think there's one that can be found at the root of it. This will be a long article, so please bear with me.
+
+## Introduction
+
+First of all, I believe there is an all-too-common misunderstanding about the very nature of PrestaShop. Many people think of PrestaShop as a product: something that you download and install to build and run your shop from your browser. But it's not. Or at least I don't see it that way.
+
+When I meet people and tell them what I do for a living, I usually describe PrestaShop as an _open-source platform for developing e-commerce websites_. This is in my view, a more accurate description of what this project is about.
+
+Yes, I have met people who download it and use it as-is, but I think they belong to a declining minority. To my best understanding, most people who are choosing to use the PrestaShop project in 2019 (and who succeed in doing so) do it to kickstart their own development. They are usually agencies or experts, and often they are also part of our rich community of module and theme developers who sell on the [Addons Marketplace][addons].
+
+This means that we, the people who work on PrestaShop's Core code, have to keep tree kinds of "users" into account:
+
+1. The shoppers that buy things on PrestaShop shops,
+2. ... the merchants that run PrestaShop shops (and ultimately, pay for),
+3. ... and the **developers** that work using PrestaShop to build sites, or develop modules and themes.
+
+This third "user", is very, _very_ important to us. [I mean it like Steve Ballmer](https://www.youtube.com/watch?v=1VgVJpVx9bc).
+
+It's not just because we _love_ developers (incidentally, we do!). It's because PrestaShop cannot survive without them. As such, we want to encourage them to develop for our platform. 
+
+As platform developers, we face a big challenge that most developers (or at least, web developers) usually don't need to care about: **backwards compatibility**.
+
+Developers who work with PrestaShop (or any platform for that matter) expect things to work in a certain, predictable way. Either because the vendor explicitly described them that way through documentation, or because they spent time figuring out how it works on their own, in many cases through excruciating trial-and-error and reverse engineering of undocumented code. The bottom line is, once they get something working, developers expect it to _continue working that way_—and it's a natural thing to do.
```

Comment:
Maybe we could provide an example ? Maybe one of the recent BC changes we introduced for security reasons ? I feel this statement is a little abstract.

Also, we could quote SemVer as one standard practice to manage breaking changes cycles, however maybe you did not include it because it would make the article heavier and longer.

Reply:
For SemVer: maybe then we can quote it with a reference

```
The bottom line is, once they get something working, developers expect it to _continue working that way_—and it's a natural thing to do (see SemVer*).
```

Reply:
I speak about SemVer later in the article. Do you think it's worth it mentioning it here as well?

Reply:
No, you're right